File: src/main/java/my/study/repository/design/pattern/observe/java/Subject.java
package my.study.repository.design.pattern.observe.java;

/**
 * @author Djh on 2018/7/27 14:49
 * E-Mail ：1544579459@qq.com
 */
public interface Subject {
    void register(Observe observe);

    void remove(Observe observe);

    void notifyChange();
}

File: src/main/java/my/study/repository/design/pattern/observe/kotlin/Subject.kt
package my.study.repository.design.pattern.observe.kotlin

/**
 * @author djh on  2018/10/10 17:00
 * @E-Mail 1544579459@qq.com
 */
interface Subject {
    val list: MutableList<Observe>
    
    fun registerObserve(observe: Observe)
    
    fun unRegisterObserve(observe: Observe)
    
    fun notifyDataChange()
}